pgmpy/models/MarkovModel.py

```diff
@@ -226,6 +218,8 @@ def check_model(self):
                 if cardinalities[variable] != cardinality:
                     raise ValueError(
                         'Cardinality of variable {var} not matching among factors'.format(var=variable))
+                if len(self.nodes()) != len(cardinalities):
```

Comment:
@lohani2280 The PR looks good. Could you just add a new test for these two new lines ?

Reply:
@ankurankan Yes, i'll do it very shortly :)